verifier/verifier_test.go

```diff
@@ -605,3 +605,33 @@ func TestVerifyX509TrustedIdentities(t *testing.T) {
 		})
 	}
 }
```

Comment:
Missing test for `processSignature` function.

Reply:
Added new function with name TestVersionCompatibility that calls processSignature function but checking the version compare functionality mainly